service/ExchangeService.js

```diff
@@ -7,19 +7,31 @@ var btcPrice   = null;
 var krwUsd     = null;
 var log = new Logger(AppConfig.logLevel);
 
-
+var handleError = function(message, exchange, errortype){
+	var errorObject = new Error();
+		errorObject.message = message;
+		errorObject.exchange = exchange;
+		errorObject.errorType = errortype;
+	return errorObject;
+}
 var fetchExchangeData = function(exchange, url, callback){
 	request.get(url, function (err, response, body) {
 		if ( !err && response.statusCode == 200 ){
 			try {
 				callback(null, JSON.parse(body));
-			}catch (e) {
-				return callback('ERROR parsing response from ' + exchange + '. Details: ' + e + '\nResponse::' + body);
+			}catch (e){
+				var parseError = 'ERROR parsing response from ';
```

Comment:
Actually here you are catching not just JSON.parse error, but callback possible errors also.
You can parse JSON to a variable in try-catch block and then pass to the callback:
```js
try {
  var data = JSON.parse(body);
} catch (e) {
  ...
  callback(error);
  return;
}

callback(null, data);
```